src/components/navigation/breadcrumb/Breadcrumb.js

```diff
@@ -16,10 +16,21 @@ function Breadcrumb({ children, ...props }) {
 }
 
 Breadcrumb.propTypes = {
-  children: PropTypes.oneOfType([
-    PropTypes.element,
-    PropTypes.arrayOf(PropTypes.element)
-  ])
+  children: (props, propName, component) => {
+    if (
+      PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.arrayOf(PropTypes.element)
+      ])
+    ) {
+      return props[propName].every(child => child.props.name)
+        ? null
+        : new Error('The children must have a name attribute');
+    }
+    return new Error(
+      'Please use an element as a child, and ensure the child has a name property'
+    );
+  }

```

Comment:
This looks horribly messy when using the fat arrow. It isn't easy to parse the if case or the ternary. It may also help to consider moving this whole validation function into its own module so it can be tested individually.